### PR TITLE
Feat: denied callback for push primer

### PR DESCRIPTION
**Pull request by CTLalit**
- added callback in case of cancel clicked in settings alert
- https://wizrocket.atlassian.net/browse/SDK-3670

### CODE REVIEW

File: clevertap-core/src/main/java/com/clevertap/android/sdk/PushPermissionManager.java [hunk 1]
```diff
@@ -99,6 +99,7 @@ public void showFallbackAlertDialog() {
             return Unit.INSTANCE;
         }, () -> {
             if (activity instanceof InAppNotificationActivity) {
+                ((InAppNotificationActivity) activity).notifyPermissionDenied();
```
Comment (Anush-Shand) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/PushPermissionManager.java:102]: Can `pushPermissionResultCallback` be passed to `showFallbackAlertDialog()` and hence we don't need a new function in the `InAppNotificationActivity.java`

Comment (CTLalit) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/PushPermissionManager.java:102]: extending from the current design of passing function lambdas, that is the pattern used in other cases as well.

Comment (Anush-Shand) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/PushPermissionManager.java:102]: Yess. Understood, I was thinking something like this

    ```
public void showFallbackAlertDialog(InAppNotificationActivity.PushPermissionResultCallback pushPermissionResultCallback) {
        AlertDialogPromptForSettings.show(activity, () -> {
            Utils.navigateToAndroidSettingsForNotifications(activity);
            isFromNotificationSettingsActivity = true;
            return Unit.INSTANCE;
        }, () -> {
            if (activity instanceof InAppNotificationActivity) {
                ((InAppNotificationActivity) activity).didDismiss(null);
                pushPermissionResultCallback.onPushPermissionDeny();
            }
            return Unit.INSTANCE;
        });
    }
```

Comment (piyush-kukadiya) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/PushPermissionManager.java:102]: @CTLalit @Anush-Shand Does `onResume()` of `InAppNotificationActivity` gets called when dialog is cancelled? If yes then we can reuse this LOC  `isFromNotificationSettingsActivity = true;` which is in `onAccept` of dialog. In that case  `onResume()` of `InAppNotificationActivity` handles notifying deny call but there will be extra if checks in `onResume()` also `isFromNotificationSettingsActivity` naming need to changed.

**Review by piyush-kukadiya — COMMENTED** [review]
Added comments

File: clevertap-core/src/main/java/com/clevertap/android/sdk/PushPermissionManager.java (hunk 1 above)
Comment (CTLalit) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/PushPermissionManager.java:102]: onResume() of InAppNotificationActivity does not get called in this flow as we discussed in the call

**Review by piyush-kukadiya — APPROVED** [review]
LGTM

**Pull request merged**